src/structs.rs

```diff
@@ -107,6 +107,63 @@ impl From<&AssetCode> for BaseField {
     }
 }
 
+/// Amount value type
+#[tagged_blob("AMOUNT_VALUE")]
+#[derive(
+    Debug,
+    Clone,
+    Copy,
+    PartialEq,
+    Default,
+    Hash,
+    Eq,
+    From,
+    Into,
+    Add,
+    AddAssign,
+    Sub,
+    SubAssign,
+    Mul,
+    MulAssign,
+    Div,
+    DivAssign,
+    Rem,
+    RemAssign,
+    PartialOrd,
+    Ord,
+    Sum,
+)]
+pub struct AmountValue(pub(crate) u128);
```

Comment:
Should have `Display` and `FromStr` impls (maybe also `LowerHex`) (currently implied by `tagged_blob`, but I recommend replacing `tagged_blob` above)